chia/_tests/wallet/conftest.py

```diff
@@ -197,3 +209,89 @@ async def wallet_environments(
+                    launcher_id = eve_coin.parent_coin_info
+                    vault_root = VaultRoot.from_bytes(launcher_id)
+
+                    await wallet_environments.process_pending_states(
+                        [
+                            WalletStateTransition(
+                                pre_block_balance_updates={
+                                    1: {
+                                        "init": True,
+                                        "set_remainder": True,
+                                    }
+                                },
+                                post_block_balance_updates={
+                                    1: {
+                                        "set_remainder": True,
+                                    }
+                                },
+                            )
+                            for _ in wallet_environments.environments
+                        ]
+                    )
+                    await env.node.keychain_proxy.add_key(launcher_id.hex(), label="vault", private=False)
+                    await env.restart(vault_root.get_fingerprint())
+                    await wallet_environments.full_node.wait_for_wallet_synced(env.node, 20)
+                    assert isinstance(env.xch_wallet, Vault)  # mypy
+                    await env.xch_wallet.sync_vault_launcher()
+
+                for env, blocks_needed in zip(wallet_environments.environments, request.param["blocks_needed"]):
+                    if blocks_needed > 0:
+                        assert isinstance(env.xch_wallet, Vault)  # mypy
+                        farm_to = env.xch_wallet.get_p2_singleton_puzzle_hash()
+                        await wallet_environments.full_node.farm_blocks_to_puzzlehash(
+                            count=blocks_needed,
+                            farm_to=farm_to,
+                            guarantee_transaction_blocks=True,
+                        )
+                        await full_node[0]._api.wait_for_wallet_synced(wallet_node=service._node, timeout=20)
+                        await wallet_environments.process_pending_states(
+                            [
+                                WalletStateTransition(
+                                    pre_block_balance_updates={
+                                        1: {
+                                            "init": True,
+                                            "set_remainder": True,
+                                        }
+                                    },
+                                    post_block_balance_updates={
+                                        1: {
+                                            "set_remainder": True,
+                                        }
+                                    },
+                                )
+                                for _ in wallet_environments.environments
+                            ]
+                        )
```

Comment:
Don't we already have logic to farm blocks to a puzzle hash in this framework? https://github.com/Chia-Network/chia-blockchain/blob/68e587124715e0a705219b02efbab2f1ade82769/chia/_tests/wallet/conftest.py#L180-L184